File: BOJ/[BOJ 11659] 구간 합 구하기 4.py
import sys
input = sys.stdin.readline


n, m = map(int, input().split())
lst = list(map(int, input().split()))
for i in range(1, n):
    lst[i]+=lst[i-1]
result = [0]*m
for j in range(m):
    l, r = map(int, input().split())
    if l>1:
        result[j] = lst[r-1]-lst[l-2]
    else:
        result[j] = lst[r-1]
for k in range(m):
    print(result[k])

File: BOJ/[BOJ 11659] 구간합 구하기 4.js
const fs = require("fs");
const input = fs
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n")
  .map((i) => i.split(" ").map(Number));

const [N, M] = input.splice(0, 1)[0];
const arr = input.splice(0, 1)[0];
const lst = input;

// 누적합 구하기
const prefixSum = Array.from({ length: N }, () => 0);
let total = 0;
for (let i = 0; i < N; i++) {
  total += arr[i];
  prefixSum[i] = total;
}

// 구간합 구하기
const results = [];
for (const [i, j] of lst) {
  if (i > 1) {
    results.push(prefixSum[j - 1] - prefixSum[i - 2]);
  } else {
    results.push(prefixSum[j - 1]);
  }
}
console.log(results.join("\n"));
